src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,209 @@
+package duke;
+import duke.command.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    protected static ArrayList<Task> tasks = new ArrayList<Task>();
+    private static final String FILE_PATH = "data/duke.txt";
+    public static void main(String[] args) {
+        printGreetings();
+        try {
+            getTasksFromFile();
+        } catch (IOException e) {
+            System.out.print("Something went wrong: \" + e.getMessage()");
+        }
+        executeCommand(tasks);
+        writeToFile();
+        printGoodbye();
+    }
+
+    private static void writeToFile() {
+        StringBuilder textToWrite = new StringBuilder();
+        for (Task task : tasks) {
+            textToWrite.append(task.getTag()).append(" | ").append(task.getStatusBinary()).append(" | ").append(task.getTaskName());
+            if (task instanceof Event || task instanceof Deadline) {
+                textToWrite.append(" | ").append(task.getDetails());
+            }
+            textToWrite.append(System.lineSeparator());
+        }
+        try {
+            IOManager.writeToFile(FILE_PATH, textToWrite.toString());
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void getTasksFromFile() throws IOException {
+        try {
+            IOManager.readFileContents(FILE_PATH);
+            System.out.println("File found!");
+        } catch (FileNotFoundException e) {
+            File f = new File(FILE_PATH);
+            f.createNewFile();
+        }
+    }
+
+    private static void executeCommand(ArrayList<Task> tasks) {
```

Comment:
Could consider extracting some of the code to another method as executeCommand has more than 30 lines of code (long method).